script.js

```diff
@@ -0,0 +1,57 @@
+const numbers = document.querySelectorAll('.number');
+const operations = document.querySelectorAll('.operation');
+const decimal = document.getElementById('dot');
+const clearButtons = document.querySelectorAll('.clear');
+const display = document.getElementById('display');
+let memoryCurrentNumber = 0;
+let memoryNewNumber = false;
+let memoryPendingOperation = '';
+
+numbers.forEach((e, i) => numbers[i].addEventListener('click', (e) => operandEnter(e.target.textContent)));
+
+let operandEnter = (number) => {
+  if (memoryNewNumber) {
+    display.value = number;
+    memoryNewNumber = false;
+  } else display.value === '0' ? display.value = number : display.value += number;
+  display.value = display.value.substring(0,10);
+};
+
+operations.forEach((e, i) => operations[i].addEventListener('click', (e) => operatorEnter(e.target.textContent)));
+
+let operatorEnter = (op) => {
+  let localOperationMemory = display.value;
+  if (memoryNewNumber && memoryPendingOperation !== '=') display.value = memoryCurrentNumber;
+    else { memoryNewNumber = true;
+    if (memoryPendingOperation === '+') memoryCurrentNumber += +localOperationMemory;
+      else if (memoryPendingOperation === '-') memoryCurrentNumber = (memoryCurrentNumber*Math.pow(10, 9) - (+localOperationMemory*Math.pow(10, 9)))/Math.pow(10, 9);
+      else if (memoryPendingOperation === '*') memoryCurrentNumber *= +localOperationMemory;
+      else if (memoryPendingOperation === '/') memoryCurrentNumber /= +localOperationMemory;
```

Comment:
For such cases use better switch-case construction.

Reply:
DONE!